docs/source/conf.py

```diff
@@ -102,9 +102,13 @@ def _get_date():
     "sphinx_copybutton",
     "sphinx_toggleprompt",
     "sphinxcontrib.spelling",
-    "sphinx_gallery.gen_gallery",
+    
 ]
 
+DOCS_VERSION = os.environ.get('DOCS_VERSION', 'none')
+
+if DOCS_VERSION in ['dev-recipes', 'latest', 'archive']:
+    extensions.append('sphinx_gallery.gen_gallery')
```

Comment:
I just thought that `DOCS_VERSION` seemed a bit generic, and might even be used by some other application, perhaps.

```suggestion
CF_DOCS_MODE = os.environ.get('CF_DOCS_MODE', 'none')

if CF_DOCS_MODE in ['dev-recipes', 'latest', 'archive']:
    extensions.append('sphinx_gallery.gen_gallery')
```